src/main/java/seedu/address/model/person/PersonFactory.java

```diff
@@ -0,0 +1,207 @@
+package seedu.address.model.person;
+
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DONATED_AMOUNT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_HOURS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PARTNERSHIP_END_DATE;
+
+import java.util.Set;
+
+import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * A factory class to create instances of {@code Person} subclasses based on the specified {@code Role}.
+ * <p>
+ * This factory uses the provided role and other person attributes to determine which type of
+ * {@code Person} subclass to instantiate (e.g., {@code Volunteer}, {@code Donor}, or {@code Partner}).
+ * <p>
+ * Each role requires specific attributes that are checked during the creation process. If a required
+ * attribute is missing or invalid, a {@code ParseException} is thrown. The factory relies on the
+ * {@code ArgumentMultimap} to supply role-specific fields.
+ */
+public class PersonFactory {
+
+    /**
+     * Creates a specific subclass of {@code Person} based on the provided role-specific fields.
+     * This method examines the provided fields to determine which subclass of {@code Person}
+     * (i.e., {@code Volunteer}, {@code Donor}, or {@code Partner}) to instantiate.
+     *
+     * @param name The name of the person.
+     * @param phone The phone number of the person.
+     * @param email The email address of the person.
+     * @param address The address of the person.
+     * @param tags The tags associated with the person.
+     * @param hours The hours contributed, applicable for {@code Volunteer}; otherwise, {@code null}.
+     * @param donatedAmount The donated amount, applicable for {@code Donor}; otherwise, {@code null}.
+     * @param partnershipEndDate The partnership end date, applicable for {@code Partner}; otherwise, {@code null}.
+     * @return A new instance of the appropriate {@code Person} subclass based on the provided fields.
+     * @throws IllegalArgumentException If none of the subclass-specific fields are provided, or if
+     *         more than one subclass-specific field is non-null.
+     */
+    public static Person createPerson(Name name, Phone phone, Email email, Address address, Set<Tag> tags,
+                                      Hours hours, DonatedAmount donatedAmount, Date partnershipEndDate) {
+        if (hours != null) {
+            return new Volunteer(name, phone, email, address, tags, hours);
+        } else if (donatedAmount != null) {
+            return new Donor(name, phone, email, address, tags, donatedAmount);
+        } else if (partnershipEndDate != null) {
+            return new Partner(name, phone, email, address, tags, partnershipEndDate);
```

Comment:
I am not sure if I am understanding this right but why are there 2 `createPerson` methods?

Reply:
There are different scenarios where createPerson is required. For example, in an add command, you should be passing in the argmultimap. In edit command, it is necessary to pass in personToEdit. That's why it is overloaded

Reply:
If that is the case, I am not quite convinced this is the best way to do it? It does seem quite confusing to have different methods for doing same thing.

Not sure if its applicable for this case, but for instance in the `EditCommand` class, there could be parameters passed in, or not passed in, which are handled by the internal workings of the method. I think the important thing is that it allows the workings of `parse` method in `EditCommand` to be abstracted away.

In my opinion, having multiple `createPerson` methods with different arguments that can be passed in, weakens the abstraction barrier in a sense. 

Not sure how this can be improved upon. Let me know what you think and feel free to chime in as well
@voldiemort @chongsun2002 @Justin-Yeo

Reply:
I initially designed this to SLAP more in CommandParse classes. 

In your example, you would have to parse the values in EditCommandParse before calling createPerson. But at this moment you do not know what specific type of person you are dealing with, thus switch statements and type casting would be needed to parse necessary values. This would probably make the code a lot messier. Just some of  my thoughts :)

Reply:
But I do think that further abstractions could be made. Will definitely look into this in v1.4